#!/usr/bin/env python
PACKAGE = "fusion_estimator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("EstimatorPar1", double_t, 0, "Parameter 1", 0, -1, 1)
gen.add("EstimatorPar2", double_t, 0, "Parameter 2", 0, -1, 1)
gen.add("EstimatorPar3", double_t, 0, "Parameter 3", 0, -1, 1)
gen.add("EstimatorPar4", double_t, 0, "Parameter 4", 0, -1, 1)
gen.add("EstimatorPar5", double_t, 0, "Parameter 5", 0, -1, 1)
gen.add("EstimatorPar6", double_t, 0, "Parameter 6", 0, -1, 1)
gen.add("EstimatorPar7", double_t, 0, "Parameter 7", 0, -1, 1)
gen.add("EstimatorPar8", double_t, 0, "Parameter 8", 0, -1, 1)
gen.add("EstimatorPar9", double_t, 0, "Parameter 9", 0, -1, 1)
gen.add("EstimatorPar10", double_t, 0, "Parameter 10", 0, -1, 1)
gen.add("EstimatorPar11", double_t, 0, "Parameter 11", 0, -1, 1)
gen.add("EstimatorPar12", double_t, 0, "Parameter 12", 0, -1, 1)
gen.add("EstimatorPar13", double_t, 0, "Parameter 13", 0, -1, 1)
gen.add("EstimatorPar14", double_t, 0, "Parameter 14", 0, -1, 1)
gen.add("EstimatorPar15", double_t, 0, "Parameter 15", 0, -1, 1)
gen.add("EstimatorPar16", double_t, 0, "Parameter 16", 0, -1, 1)
gen.add("PositionX", double_t, 0, "PositionX", 0, -10, 10)
gen.add("PositionY", double_t, 0, "PositionY", 0, -10, 10)
gen.add("PositionZ", double_t, 0, "PositionZ", 0, -10, 10)
gen.add("ResetFlag", bool_t, 0, "Initiate", False)

exit(gen.generate(PACKAGE, "fusion_estimator", "Estimator"))